internal/services/iotcentral/parse/nested_item.go

```diff
@@ -0,0 +1,145 @@
+package parse
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type NestedItemObjectType string
+
+const (
+	baseUrlTemplate                                 = "https://%s.azureiotcentral.com"
+	NestedItemTypeOrganization NestedItemObjectType = "organizations"
+)
+
+// PossibleNestedItemObjectTypeValues returns a string slice of possible "NestedItemObjectType" values.
+func PossibleNestedItemObjectTypeValues() []string {
+	return []string{string(NestedItemTypeOrganization)}
+}
+
+var _ resourceids.Id = NestedItemId{}
+
+type NestedItemId struct {
+	IotcentralBaseUrl string
+	SubDomain         string
+	NestedItemType    NestedItemObjectType
+	Id                string
+}
+
+func NewNestedItemID(iotcentralBaseUrl string, nestedItemType NestedItemObjectType, id string) (*NestedItemId, error) {
+	iotCentralUrl, err := url.Parse(iotcentralBaseUrl)
+	if err != nil || iotcentralBaseUrl == "" {
+		return nil, fmt.Errorf("parsing %q: %+v", iotcentralBaseUrl, err)
+	}
+
+	// in case the user has provided a port, we need to remove it
+	if hostParts := strings.Split(iotCentralUrl.Host, ":"); len(hostParts) > 1 {
+		iotCentralUrl.Host = hostParts[0]
+	}
+
+	subDomain, err := parseSubDomain(iotcentralBaseUrl)
+	if err != nil {
+		return nil, fmt.Errorf("parsing %q: %+v", iotcentralBaseUrl, err)
+	}
+
+	return &NestedItemId{
+		IotcentralBaseUrl: iotCentralUrl.String(),
+		SubDomain:         subDomain,
+		NestedItemType:    nestedItemType,
+		Id:                id,
+	}, nil
+}
+
+func (id NestedItemId) ID() string {
+	// example: https://subdomain.baseDomain/api/type/fdf067c93bbb4b22bff4d8b7a9a56217
+	segments := []string{
+		strings.TrimSuffix(id.IotcentralBaseUrl, "/"),
+		"api",
+		string(id.NestedItemType),
+		id.Id,
+	}
+
+	return strings.TrimSuffix(strings.Join(segments, "/"), "/")
+}
```

Comment:
Whilst we do this "Nested Item ID" in Key Vault (for historical reasons), we're actually moving away from this pattern in favour of having more specific ID parsers (e.g. an `OrganizationID` parser) to avoid needing to specify the Nested Item type in various places (we'll be updating the other usages of this pattern in other parts of the provider at some point fwiw) - so rather than doing this could we make this specific to Organizations:

```suggestion
func (id OrganisationId) ID() string {
	return fmt.Sprintf("https://%s.%s/api/organizations/%s", id.SubDomainName, id.DomainSuffix, id.OrganizationID)
}
```